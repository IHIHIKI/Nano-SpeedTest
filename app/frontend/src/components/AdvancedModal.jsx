import React from 'react';
import Modal from 'react-modal';

export default (props) => (
    <Modal
        appElement={document.getElementById('root')}
        isOpen={props.open}
        contentLabel='Advanced Settings'
    >
        <h3>Advanced Settings</h3>
        <hr/>
        <button className='btn btn-primary' onClick={props.handleAdvancedSettings}>Submit</button>
    </Modal>
);