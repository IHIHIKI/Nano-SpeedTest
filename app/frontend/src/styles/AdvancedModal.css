.adv-form-button {
    margin-bottom: 1rem;
}

#submit {
    margin-right: 10px;
}