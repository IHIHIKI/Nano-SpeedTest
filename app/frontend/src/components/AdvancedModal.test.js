import React from 'react';
import renderer from 'react-test-renderer';
import AdvancedModal from 'components/AdvancedModal';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter} from 'react-router-dom';
import { shallow } from 'enzyme';

const mockStore = configureStore();
let store = mockStore({
    navigation: {
        activeTab: ''
    }
});

it('shallow renders without crashing', () => {
    shallow(<AdvancedModal store={store}/>);
});

it(`renders base snapshot correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <AdvancedModal />
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it(`renders two cities snapshot correctly`, () => {
    const state = {
        locationsActive: true,
    };
    const props = {
        settings: {
            advSettings: {
                values: {
                    origin: 'A',
                    destination: 'B'
                }
            }
        },
        nodes: {
            123: {
                id: 123,
                location: 'A'
            },
            222: {
                id: 222,
                location: 'B'
            }

        }
    };
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <AdvancedModal state={state} {...props}/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it(`renders multi snapshot correctly`, () => {
    const state = {
        multipleActive: true,
    };
    const props = {
        nodes: {
            123: {
                id: 123,
                location: 'A'
            },
            222: {
                id: 222,
                location: 'B'
            }

        }
    };
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <AdvancedModal state={state} {...props}/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});