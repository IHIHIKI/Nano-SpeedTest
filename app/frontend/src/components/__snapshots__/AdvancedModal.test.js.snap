// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders base snapshot correctly 1`] = `null`;

exports[`renders multi snapshot correctly 1`] = `null`;

exports[`renders two cities snapshot correctly 1`] = `null`;
